### ISSUE TITLE
Wrong offset value from VTimeZone

### ISSUE THREAD

**Issue by nguyenanduong** [ical4j/ical4j#260]
**Describe the bug**
Generating event 

**To Reproduce**
1. This test fails

        DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm");
        DateTime dateTime =new DateTime(df.parse("2018-03-11 00:00"));
        dateTime.setTimeZone(TimeZoneRegistryFactory.getInstance().createRegistry().getTimeZone("America/Los_Angeles"));

        Calendar cal = Calendar.getInstance(dateBeforeDST.getTimeZone()); 
        cal.setTime(dateTime);
        cal.set(Calendar.HOUR_OF_DAY, 2);

        assertThat(cal.getTime().getHours(), is(3));

2. Modify the test and observe that it passes

        DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm");
        DateTime dateTime =new DateTime(df.parse("2018-03-11 00:00"));
        dateTime.setTimeZone(TimeZoneRegistryFactory.getInstance().createRegistry().getTimeZone("America/Los_Angeles"));

        Calendar cal = Calendar.getInstance(TimeZone.getTimeZone("America/Los_Angeles")); 
        cal.setTime(dateTime);
        cal.set(Calendar.HOUR_OF_DAY, 2);

        assertThat(cal.getTime().getHours(), is(3));

**Expected behavior**
Test pass.

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Windows, Linux, Android]
 - Java Version [e.g. 7, 8, 9]

**Additional context**

**Comment by nguyenanduong** [comment]
Any update on this? Thanks

**Comment by benfortuna** [comment]
Apologies for delayed response. 

Can you provide more info on the nature of the test failure (what is the result), and the value of `dateBeforeDST`.

Also if you could explain what the code is doing it would be appreciated. Appears that you are creating a `java.util.Calendar` based on a `DateTime`, which may not be of much value ultimately.

**Comment by nguyenanduong** [comment]
Hi Ben,
I dug into iCal4J and reported the nearly root cause. Actually, the problem can be seen while generating dates from an RRule which occur right at DST change. For example: 

ZoneId timezone = ZoneId.of("America/Sao_Paulo");
TimeZone icalTZ = icalTimeZoneRegistry.getTimeZone(timezone);

DateTime seed = new DateTime(DateTimeFormatter.ofPattern(ICAL_DEFAULT_DATETIME_FORMAT).format(LocalDateTime.parse("2018-10-07T00:00:00")), icalTZ);

DateTime until = new DateTime(DateTimeFormatter.ofPattern(ICAL_DEFAULT_DATETIME_FORMAT).format(LocalDateTime.parse("2019-01-01T00:00:00")), icalTZ);

final Recur recur = new Recur("FREQ=WEEKLY;BYDAY=SU;INTERVAL=1 ");
DateList list = recur.getDates(seed, until, Value.DATE_TIME);

Notice that the list contains an occurrence of 2018-11-03T23:00:00 which is unexpected. Midnight 11-04 is DST change in Sao_Paulo, so I expect an occurrence at 2018-11-04T01:00:00 (midnight)

Please let me know if you need more information
Thanks.

**Comment by SirGlorg** [comment]
It seems that this comes down to the ics definitions, which define DST recurrent rule for the "america/sao_paulo" starting 2018 onwards (if I'm reading that right). I commented on https://github.com/benfortuna/tzurl/issues/34 to see if this can be re-generated/fixed.

Sadly, the tzurl repository contains a docker build machine, however, when I built it and ran, the generated Sao Paulo timezone definition remained the same. I think that it needs to be fixed there?
